packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/TransactionModal.tsx

```diff
@@ -0,0 +1,227 @@
+import { Toast } from "#/components/Toast";
+import { PoolMetadataContext } from "#/contexts/PoolMetadataContext";
+import { TransactionStatus, useTransaction } from "#/hooks/useTransaction";
+
+type Stage = {
+  pinColor: string;
+  confirmColor: string;
+  writeColor: string;
+};
+
+const STAGE_MAPPING: Record<TransactionStatus, Stage> = {
+  [TransactionStatus.PINNING]: {
+    pinColor: "border-yellow-400 text-yellow-400",
+    confirmColor: "bg-gray-400",
+    writeColor: "border-gray-400 text-gray-400",
+  },
+  [TransactionStatus.CONFIRMING]: {
+    pinColor: "border-green-400 text-green-400",
+    confirmColor: "bg-green-400",
+    writeColor: "border-yellow-400 text-yellow-400",
+  },
+  [TransactionStatus.SUBMITTING]: {
+    pinColor: "border-green-400 text-green-400",
+    confirmColor: "bg-green-400",
+    writeColor: "border-yellow-400 text-yellow-400",
+  },
+  [TransactionStatus.CONFIRMED]: {
+    pinColor: "border-green-400 text-green-400",
+    confirmColor: "bg-green-400",
+    writeColor: "border-green-400 text-green-400",
+  },
+  [TransactionStatus.PINNING_ERROR]: {
+    pinColor: "border-red-400 text-red-400",
+    confirmColor: "bg-gray-400",
+    writeColor: "border-gray-400 text-gray-400",
+  },
+  [TransactionStatus.WRITE_ERROR]: {
+    pinColor: "border-green-400 text-green-400",
+    confirmColor: "bg-green-400",
+    writeColor: "border-red-400 text-red-400",
+  },
+  [TransactionStatus.AUTHORIZING]: {
+    pinColor: "border-yellow-400 text-yellow-400",
+    confirmColor: "bg-gray-400",
+    writeColor: "border-gray-400 text-gray-400",
+  },
+  [TransactionStatus.WAITING_APPROVAL]: {
+    pinColor: "border-green-400 text-green-400",
+    confirmColor: "bg-green-400",
+    writeColor: "border-yellow-400 text-yellow-400",
+  },
+};
+
+function ConfirmationAlert({
+  handleSubmit,
+}: {
+  handleSubmit: (value: boolean) => void;
+}) {
+  return (
+    <div className="mt-4 px-1">
+      <div className="w-full space-y-6">
+        <h1 className="my-2 text-xl text-yellow-400">Are you sure?</h1>
+        <span className=" text-gray-200">
+          This procedure will cost gas fee. You make sure there is no data to
+          change before updating.{" "}
+        </span>
+      </div>
+      <div className="mt-7 flex gap-5">
+        <DialogPrimitive.Close asChild>
+          <Button className="border border-gray-100 text-gray-100">
+            Cancel
+          </Button>
+        </DialogPrimitive.Close>
+        <Button
+          onClick={() => handleSubmit(true)}
+          className="w-full bg-yellow-400 text-gray-900 hover:bg-yellow-300 focus-visible:bg-yellow-300 disabled:bg-yellow-200"
+        >
+          I'm sure
+        </Button>
+      </div>
+    </div>
+  );
+}
+
+function Ball({ color, children }: PropsWithChildren<{ color: string }>) {
+  return (
+    <div
+      className={cn(
+        "w-8 h-8 rounded-full flex items-center justify-center",
+        "font-bold text-lg border-[1px]",
+        color
```

Comment:
don't abstract it if you don't need
```suggestion
        classNames
```